Commonalities/documentation/API-design-guidelines.md

```diff
@@ -1021,3 +1027,180 @@ Finally, this part describes the OAuth security applied to the API. This spec is
+A notification is the publication at the listener address about an occurred event.
+Event type managed are explicitly defined in CAMAPA API OAS.
+
+### 12.1 Subscription
+
+We distinguish 2 types of subscription:
+- Instance-based subscription (indirect creation)
+- Resource-based subscription (direct creation)
+
+**Instance-based subscription**
+
+An instance-based subscription is a subscription indirectly created, additionnally, to another resource creation. For example for a Payment request (in Carrier Billing API), in the `POST/payments`, the API consumer could request to get event notification about **this** Payment request processing update. The subscription is not an entity and its lifecycle is linked to the managed entity (the Payment resource in this case). The subscription terminates with the managed entity.
+
+Providing this capability is optionnal for any CAMARA APIs depending on UC requirements
+
+If this capability is present in CAMARA API, following attributes **must** be used in the POST request for the managed entity:
+
+| attribute name | type | attribute description | cardinality |
+| ----- |	-----  |	 -----  | -----  | 
+| notificationUrl | string |callback address where the notification must be POST-ed | mandatory |
+| notificationAuthToken | string | authentification token for callback API | optional |
+
+
+**Resource-based subscription**
+
+A resource-based subscription is is subscription managed as a resource. A dedicated endpoint is provided to request subscription creation.  As this subscription is managed API resource, it is identified and operations to search, retrieve and delete them must be provided.
+
+Note: It is perfectly valid for a CAMARA API to have several subscriptions endpoints when distinct events are managed (for example in Device status API we manage distinct subscription for roaming status and reachability status event)
+
+In order to ease developer adoption, the pattern for Resource-based subscription must be the consistent for all API providing this feature.
+
+4 operations must be defined:
+
+| operation | path | description |
+| ----- |	-----  |	 -----  | 
+| POST | `/subscriptions/<subscriptionName>` | Operation to request a subscription.  eg. POST /subscriptions/roamingStatus    |
+| GET | `/subscriptions/` | Operation to retrieve a list of subscriptions - could be an empty list.  eg. `GET /subscriptions?type=ROAMING_STATUS&ExpireTime.lt=2023-03-17` |
+| GET | `/subscriptions/{subscriptionId}` | Operation to retrieve a subscription |
+| DELETE | `/subscriptions/{subscriptionId}` | Operation to delete a subscription |
+
+Following table provides subscriptions attributes
+
+| name | type | attribute description | cardinality |
+| ----- |	-----  |	 -----  |  -----  | 
+| notificationUrl | string | callback address where the notification must be POST-ed | mandatory |
+| notificationAuthToken | string | authentification token for callback API | optional |
```

Comment:
Do we only want to support Apikey for authentication?

Reply:
As mentioned in https://github.com/camaraproject/WorkingGroups/issues/156, we suggest two things:
* Define clearly the scope of use of this "token":
   - whether it is a token used in the way-back of the communication
   - in which way (i.e. within Authorization header, being a Bearer Token -I guess-)
   - conventions of this format. Considerations about length, format, mechanism that ensure unicity/randomness
* Adding feature of notification body signing

Reply:
@akoshunyadi In first approach that was my assumption as we already have this pattern in existing API. Do you think we need to describe other alternative there? 

Fully agree @PedroDiez 
Do you want to me to update the proposal accordingly or do you prefer to handle it ? I completely open to your contribution.

Reply:
Let me compile some information first @bigludo7 in order to share in advance an open for everybody to look at it. Then we can work in the update of this PR

@akoshunyadi We also thought about this, but at the end (if understanding well your comment), to implement an oAuth-based approach it will set some requirments in API Consumer side, in order to provide some kind of Authorization Server, so we as telco operators can ask for a Token.
Think it is a requirement we should not impose to API Consumers in order to ease them the integration.

If your comment goes in other line, please comment in order to understand the proposal

Reply:
I have contributed, please take a look if it is fine with you

Reply:
Fine for me ! Thanks !!!

Reply:
@PedroDiez My idea was to provide both possibilities. So the consumer can use the API-key for simple use cases, but he would have the chance for higher level of security with of course more effort on his side.